Metadata display

jaguillette opened the pull request:
This PR addresses some issues that came up in discussion with Jinah Kim's Mapping Color team. The current metadata display is limited to just the Dublin Core metadata, and so it ignores any other metadata, including item type or other metadata added by plugins. This removes that restriction on what metadata is included in manifests, and adds a metadata element to link back to the page in Omeka.

jaguillette committed:
Add all metadata and link back to Omeka record

Previous setup adds only Dublin Core metadata, no Item Type metadata or any other type added by a plugin. This change displays all metadata, as well as a link back to the original Omeka record.

jaguillette committed:
rephrase and use translation function

Item is too narrow a term, and this should be wrapped in the translation function for public-facing strings.

jaguillette committed:
don't need to say Omeka twice

dodget commented on libraries/IiifItems/Util/Canvas.php:
```diff
@@ -244,11 +244,11 @@ public static function fileCanvasJson($file, $canvasId=null, $applyDublin=false)
         $json = self::blankTemplate($canvasId, $fileImageJson['resource']['width'], $fileImageJson['resource']['height'], $displayTitle, array($fileImageJson));
         // Apply Dublin Core metadata
         if ($applyDublin) {
```
I wonder if its worth it to change upstream variable names as well, so the flow of the script makes as much sense as it did previously? Specifically, I wonder if you should change `$applyDublin` to `$applyMetadata`? Maybe not worth it to do all of that, though.

arthurian replied:
I think this is a good idea given that it's not just applying dublin core anymore, so it might be good to update that variable/parameter to reflect the new functionality. If that's done, then also be sure to update the docblock comments that reference `@param boolean $applyDublin`.

dodget commented on libraries/IiifItems/IiifUtil.php:
```diff
@@ -72,16 +72,21 @@ protected static function addDublinCoreMetadata(&$jsonData, $record) {
+                )
+            );
+        }
+        foreach ($elements as $elementSetName => $elementSet) {
+            foreach ($elementSet as $elementName => $elementContent) {
+                $jsonData['metadata'][] = array(
+                    'label' => $elementName,
+                    'value' => join('<br/>', $elementContent)
+                );
```
I'm not sure if I'm reading this right, but `$jsonData['metadata']` is an array of associated arrays (dicts)? If so, I find the `$jsonData['metadata'][] =` difficult to read, but maybe thats because I'm a php novice. I would find this much easier to understand:
```php
$elementArray = array(
    'label' => $elementName,
    'value' => join('<br/>', $elementContent)
);
array_push($jsonData['metadata'], $elementArray);
```
But again, I may be reading that wrong?

arthurian replied:
I think you're reading that right @dodget. For better or worse, using empty square brackets to append an item onto an array is considered idiomatic in PHP. According to the docs, the [array_push()](https://www.php.net/array_push) function is preferred when adding multiple items onto an array at one time, as opposed to a single item, which is the case here.

dodget replied:
Thanks for the context Artie. Disregard this comment then! 👍

dodget reviewed (approved):
Looks good, just a couple of inconsequential thoughts 👍

arthurian commented on libraries/IiifItems/IiifUtil.php:
```diff
@@ -72,16 +72,21 @@ protected static function addDublinCoreMetadata(&$jsonData, $record) {
+            foreach ($elementSet as $elementName => $elementContent) {
+                $jsonData['metadata'][] = array(
+                    'label' => $elementName,
+                    'value' => join('<br/>', $elementContent)
```
Kudos for using the XHTML form of `<br/>` as required by the IIIF spec when [HTML is used in property values](https://iiif.io/api/presentation/2.1/#html-markup-in-property-values). It's interesting that the spec seems to suggest that property values should be enclosed in start/end tags so that consuming applications can easily test if the value is plain text or HTML (e.g. `'<span>...</span>'`). I'm not sure I'd recommend doing that here if Mirador is parsing things OK as-is. But you might consider using `trim()` on the joined string to ensure there's no leading or trailing whitespace.

arthurian reviewed (approved):
Looks good to overall @jaguillette . I would just echo some of @dodget's suggestions.

jaguillette committed:
updates in response to feedback

Changed upstream function names to reflect expanding what metadata is included, and added a trim function to joined metadata to eliminated extra whitespace
